client/src/pages/Search/MapView/RestaurantPreview.js

```diff
@@ -1,7 +1,103 @@
-function RestaurantPreview({ restaurant }) {
+import React from "react";
+import ReviewStars from "../../../components/ReviewStars";
+import { IconButton } from "@mui/material";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { Link } from "react-router-dom";
+
+const classes = {
+  previewContainer: {
+    height: "120px",
+    width: "100%",
+    backgroundColor: "rgb(224,224,224)",
+    display: "flex",
+    justifyContent: "space-between",
+  },
+  imageContainer: {
+    height: "75%",
+    width: "100px",
+    margin: "1rem",
+    objectFit: "cover",
+  },
+  detailsContainer: {
+    marginLeft: "0.25rem",
+  },
+  nameContainer: {
+    fontSize: "1.5rem",
+    textDecoration: "none",
+    color: "black",
+  },
+  categoriesContainer: {
+    marginTop: "1rem",
+  },
+  reviewContainer: {
+    fontSize: "0.75rem",
+    display: "flex",
+  },
+  exitContainer: {
+    postion: "absolute",
+  },
+};
+
+function formatPhoneNumber(phoneNumberString) {
+  var cleaned = ("" + phoneNumberString).replace(/\D/g, "");
+  var match = cleaned.match(/^(1|)?(\d{3})(\d{3})(\d{4})$/);
+  if (match) {
+    var intlCode = match[1] ? "+1 " : "";
+    return [intlCode, "(", match[2], ") ", match[3], "-", match[4]].join("");
+  }
+  return null;
+}
+
+function RestaurantPreview({ restaurant, hideRestaurant }) {
+  const restaurantPage = "/restaurant?yelp_id=" + restaurant.yelp_id;
+  const phoneNumber = formatPhoneNumber(restaurant.phone);
+  const streetAddress = restaurant.address.address1 ?? "";
+  const restaurantAddress =
+    streetAddress +
+    " " +
+    restaurant.address.state +
+    ", " +
+    restaurant.address.zip_code;
```

Comment:
I think this would look cleaner:
```js
const { address1, state, zip_code } = restaurant.address;
const restaurantAddress = `${address1} ${state}, ${zip_code}`;
```

Optional: If you want to rename `address1` to `streetAddress` , you can rename it in the destructure:
```js
const { address1: streetAddress, state, zip_code } = restaurant.address;
const restaurantAddress = `${streetAddress} ${state}, ${zip_code}`;
```